examples/basic/hello-world.ts

```diff
@@ -1,71 +1,56 @@
 import {
-  Activity,
-  AgreementPoolService,
-  MarketService,
-  Package,
-  PaymentService,
-  WorkContext,
   YagnaApi,
+  MarketModuleImpl,
+  ActivityModuleImpl,
+  PaymentModuleImpl,
+  DraftOfferProposalPool,
+  WorkContext,
 } from "@golem-sdk/golem-js";
 
 (async () => {
-  console.log("Starting core services...");
-  const yagna = new YagnaApi();
-
-  const DURATION_SEC = 6 * 60;
-
-  const payment = new PaymentService(yagna);
-
-  await payment.run();
-
-  const allocation = await payment.createAllocation({
-    budget: 1.0,
-    expirationSec: DURATION_SEC,
-  });
-
-  const agreementPool = new AgreementPoolService(yagna);
-
-  const market = new MarketService(agreementPool, yagna, {
-    expirationSec: DURATION_SEC,
-  });
-
-  const workload = Package.create({
-    imageTag: "golem/alpine:latest",
-  });
-
-  await agreementPool.run();
-  await market.run(workload, allocation);
-  console.log("Core services started");
-
   try {
-    const agreement = await agreementPool.getAgreement();
-    payment.acceptPayments(agreement);
-
-    const activity = await Activity.create(agreement, yagna);
-    // Stop listening for new proposals
-    await market.end();
-
+    const yagnaApi = new YagnaApi();
+    await yagnaApi.connect();
+    const modules = {
+      market: new MarketModuleImpl(yagnaApi),
+      activity: new ActivityModuleImpl(yagnaApi),
+      payment: new PaymentModuleImpl(yagnaApi),
+    };
+    const demandOptions = {
+      demand: {
+        image: "file://golem_node_20.gvmi",
+        resources: {
+          minCpu: 4,
+          minMemGib: 8,
+          minStorageGib: 16,
+        },
+      },
+      market: {
+        rentHours: 12,
+        pricing: {
+          maxStartPrice: 1,
+          maxCpuPerHourPrice: 1,
+          maxEnvPerHourPrice: 1,
+        },
+        withProviders: ["0x123123"],
+        withoutProviders: ["0x123123"],
+        withOperators: ["0x123123"],
+        withoutOperators: ["0x123123"],
+      },
+    };
+    const invoiceFilter = () => true;
+    const debitNoteFilter = () => true;
+    const proposalPool = new DraftOfferProposalPool();
+    const proposalSubscription = await modules.market.startCollectingProposal(demandOptions, proposalPool);
+    const draftProposal = await proposalPool.acquire();
+    const agreement = await modules.market.proposeAgreement(modules.payment, draftProposal, { invoiceFilter });
+    const activity = await modules.activity.createActivity(modules.payment, agreement, { debitNoteFilter });
```

Comment:
I guess the movement of the debit note filter to agreement creation code will be done in a separate PR?

Reply:
yes